src/bun.js/api/filesystem_router.zig

```diff
@@ -211,6 +211,38 @@ pub const FileSystemRouter = struct {
         return fs_router;
     }
 
+    pub fn bustDirCacheRecursive(this: *FileSystemRouter, globalThis: *JSC.JSGlobalObject, path: []const u8) void {
+        var vm = globalThis.bunVM();
+
+        const root_dir_info = vm.bundler.resolver.readDirInfo(path) catch {
+            return;
+        };
+
+        if (root_dir_info) |dir| {
+            if (dir.getEntriesConst()) |entries| {
+                var iter = entries.data.iterator();
+                outer: while (iter.next()) |entry_ptr| {
+                    const entry = entry_ptr.value_ptr.*;
+
+                    if (entry.base()[0] == '.') {
```

Comment:
this should also ignore `node_modules`, see `router.zig`'s `banned_dirs`

Reply:
Done! I've exported the banned_dirs from router.zig and using it in the filesystemrouter

Reply:
Also there's a test included now